### ISSUE TITLE
Azure Cloud Shell Bash: Azure CLI authentication broken in azurerm v3.44.0 and newer

### ISSUE THREAD

**Issue by gui28347** [hashicorp/terraform-provider-azurerm#20653]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

 

### Terraform Version

1.3.9

### AzureRM Provider Version

3.45.0

### Affected Resource(s)/Data Source(s)

azure_resource_group

### Terraform Configuration Files

```hcl
######################################
##### CREATE-RESOURCE-GROUP MODULE terraform/modules/azure/create-resource-group/main.tf
######################################
provider "azurerm" {
    features {}
    skip_provider_registration = true
}

resource "azurerm_resource_group" "rg" {
  location = var.location
  name     = var.rgname
}


##################
###### MAIN TF terraform/testrg/main.tf
##################
terraform {
  backend "local" {
  }
}

locals {
}

provider "azurerm" {
  features {
    key_vault {
      purge_soft_delete_on_destroy = true
    }
  }
  skip_provider_registration = true
}

module "resource_group" {
  source   = "../modules/azure/create-resource-group"
  rgname   = "TESTONE"
  location = "westus3"
}
```

### Debug Output/Panic Output

```shell
## DEBUG OUTPUT
2023-02-24T22:04:13.203Z [DEBUG] provider: plugin exited
2023-02-24T22:04:13.203Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/3.45.0/linux_amd64/terraform-provider-azurerm_v3.45.0_x5 pid=1200
2023-02-24T22:04:13.201Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-02-24T22:04:13.156Z [ERROR] vertex "module.resource_group.azurerm_resource_group.rg" error: checking for presence of existing resource group: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: preparing auxiliary tokens for request: auxiliary tokens not s
upported in Cloud Shell
2023-02-24T22:04:13.083Z [ERROR] provider.terraform-provider-azurerm_v3.45.0_x5: Response contains error diagnostic: tf_proto_version=5.3 tf_provider_addr=provider tf_resource_type=azurerm_resource_group @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/di
agnostics.go:55 diagnostic_severity=ERROR diagnostic_summary="checking for presence of existing resource group: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: preparing auxiliary tokens for request: auxiliary tokens not supported in Cloud Shell" tf_req_id=
3737ff22-7fc9-4d69-0315-e624087c43cb tf_rpc=ApplyResourceChange @module=sdk.proto diagnostic_detail= timestamp=2023-02-24T22:04:13.081Z
2023-02-24T21:56:42.981Z [DEBUG] module.resource_group.azurerm_resource_group.rg: applying the planned Create change
2023-02-24T21:56:42.981Z [INFO]  Starting apply for module.resource_group.azurerm_resource_group.rg
2023-02-24T21:56:42.978Z [DEBUG] provider.terraform-provider-azurerm_v3.45.0_x5: error retrieving providers: listing Resource Providers: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: preparing auxiliary tokens for request: auxiliary tokens not supported i
n Cloud Shell. Enhanced validation will be unavailable: timestamp=2023-02-24T21:56:42.977Z
2023-02-24T21:49:11.932Z [DEBUG] provider.terraform-provider-azurerm_v3.45.0_x5: Generated Provider Correlation Request Id: GUID: timestamp=2023-02-24T21:49:11.932Z
2023-02-24T21:49:11.932Z [DEBUG] provider.terraform-provider-azurerm_v3.45.0_x5: Using client ID from Azure CLI: "GUID": timestamp=2023-02-24T21:49:11.932Z
2023-02-24T21:49:11.932Z [DEBUG] provider.terraform-provider-azurerm_v3.45.0_x5: Using default subscription ID from Azure CLI: "subscriptionID": timestamp=2023-02-24T21:49:11.932Z
2023-02-24T21:49:04.483Z [WARN]  ValidateProviderConfig from "module.resource_group.provider[\"registry.terraform.io/hashicorp/azurerm\"]" changed the config value, but that value is unused
2023-02-24T21:49:04.254Z [DEBUG] provider: using plugin: version=5
2023-02-24T21:49:04.254Z [DEBUG] provider.terraform-provider-azurerm_v3.45.0_x5: plugin address: address=/tmp/plugin1004344752 network=unix timestamp=2023-02-24T21:49:04.254Z


## REGULAR OUTPUT
module.resource_group.azurerm_resource_group.rg: Creating...
module.resource_group.azurerm_resource_group.rg: Still creating... [10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [1m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [2m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [3m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [4m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [5m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m30s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m40s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [6m50s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m0s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m30s elapsed]
╷
│ Error: checking for presence of existing resource group: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: preparing auxiliary tokens for request: auxiliary tokens not supported in Cloud Shell
│ 
│   with module.resource_group.azurerm_resource_group.rg,
│   on ../modules/azure/create-resource-group/main.tf line 6, in resource "azurerm_resource_group" "rg":
│    6: resource "azurerm_resource_group" "rg" {
```

### Expected Behaviour

Terraform script create a resource group in Azure subscription with name TESTONE.

### Actual Behaviour

terraform just hangs for about 7 minutes and ask me to confirm like usual.
After I answer: yes it hangs again for another 7 minutes and start printing the following output messages:
..
module.resource_group.azurerm_resource_group.rg: Still creating... [7m10s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m20s elapsed]
module.resource_group.azurerm_resource_group.rg: Still creating... [7m30s elapsed]
╷
│ Error: checking for presence of existing resource group: autorest/Client#Do: Preparing request failed: StatusCode=0 -- Original Error: preparing auxiliary tokens for request: auxiliary tokens not supported in Cloud Shell
│ 
│   with module.resource_group.azurerm_resource_group.rg,
│   on ../modules/azure/create-resource-group/main.tf line 6, in resource "azurerm_resource_group" "rg":
│    6: resource "azurerm_resource_group" "rg" {
│ 

Total wait time from first terraform init, around 23 minutes.

I was able to isolate the issue. The problem started with azurerm provider **3.44.0** and newer. If I use in config **3.43.0** or any older versions, I'm able to execute the same script. So definitely is something introduced in 3.44.0 that I'm not able to figure out. 

### Steps to Reproduce

1. Connect to your azure subscription using https://portal.azure.com
2. Open Azure Cloud Shell
3. Create a folder terraform
4. Create a folder terraform/modules/azure/create-resource-group
 4.1. Create a file inside this folder with name main.tf and contents are in Terraform Configuration files of this bug. (File 1)
5. Create a folder terraform/testrg
  5.1 Create a file inside this folder with name main.tf and contents are in Terraform Configuration files of this bug. (File 2)
terraform init
terraform apply

### Important Factoids

Azure Public Cloud

### References

None

**Comment by tombuildsstuff** [comment]
hi @gui28347 

This has been fixed in v3.45.0 of the Azure Provider - as such updating to this version of the provider should allow this to work, would you be able to take a look and see if that works for you?

Since this should be fixed by updating to the latest version of the Azure Provider I'm going to close this issue for the moment, but should that not work for you then please let us know and we'll take another look.

Thanks!

**Comment by gui28347** [comment]
Same problem with 3.45.0 as mentioned when I opened the issue.
